shlomobot_pytest/utils.py

```diff
@@ -77,18 +90,67 @@ def calculate_total_deducted_score(
     return min(total_points_deducted, max_points_deducted)
 
 
-def import_pyfile(module_name: str) -> ModuleType:
+def import_pyfile(py_filename: str) -> ModuleType:
     """Extracts the module from a given filename"""
-    stripped_module_name = module_name.removesuffix(".py")
+    stripped_module_name = py_filename.removesuffix(".py")
 
     return import_module(stripped_module_name)
 
 
-def get_functions_from_files(file_list: list[str]) -> list[FunctionType]:
+def get_functions_from_files(
+    file_list: list[str],
+) -> Iterator[tuple[str, FunctionType]]:
     """
     Extracts all functions from the files in file_list
     Creates a generator that returns a filename, FunctionType object pair
     """
-    for filename in file_list:
-        module = import_pyfile(filename)
-        yield filename, extract_functions(module)
+    for py_filename in file_list:
+        module = import_pyfile(py_filename)
+        for function in extract_functions(module):
+            yield function.__name__, function
+
+
+def get_clean_function_lines(function: FunctionType, should_black=True) -> list[str]:
+    """Count the amount on non comment or docstring lines in a function code"""
+    function_code = inspect.getsource(function)
+
+    # reformats file to connect split lines and remove empty lines
+    if should_black:
+        function_code = format_str(function_code, mode=FileMode(line_length=99999))
+    split_code = list(filter(None, function_code.splitlines()))
```

Comment:
Cool :D Add a comment above this line explaining that it removes empty lines (right?)
since its not obvious (I had to open the help for filter)

Reply:
it will miss any indented empty lines, no?
better would be to trim it first

```suggestion
    split_code = list(filter(lambda line: line.strip(), function_code.splitlines()))
```